packages/bootstrap-vue-next/src/components/BModal/BModal.vue

```diff
@@ -424,14 +424,16 @@ const hide = (trigger = '') => {
 // TODO: If a show is prevented, it will briefly show the animation. This is a bug
 // I'm not sure how to wait for the event to be determined. Before showing
 const showFn = () => {
-  const event = buildTriggerableEvent('show', {cancelable: true})
-  emit('show', event)
-  if (event.defaultPrevented) {
-    if (modelValue.value) modelValue.value = false
-    emit('show-prevented')
-    return
+  if (!isActive.value) {
```

Comment:
```suggestion
  if (isActive.value) return
```

Modify the logic to return early if the condition isn't met. Just makes nested lines cleaner